src/Migrations/RefToInoutMigration.hack

```diff
@@ -272,29 +317,43 @@ final class RefToInoutMigration extends BaseMigration {
     foreach (
       $root->getDescendantsOfType(FunctionCallExpression::class) as $node
     ) {
-      // Only replace calls to functions from the root namespace.
       $receiver = $node->getReceiver();
-      if ($receiver is NameToken) {
-        $fn_name = $receiver->getText();
-      } else if ($receiver is QualifiedName) {
-        $fn_name = '';
-        foreach ($receiver->getParts()->getChildren() as $part) {
-          invariant(
-            $part->getSeparator() is null ||
-              $part->getSeparator() is BackslashToken,
-            'Unexpected separator inside qualified function name: "%s"',
-            $part->getSeparatorx()->getText(),
-          );
-          $fn_name .= $part->getItem()?->getText() ?? '';
-          $fn_name .= $part->getSeparator()?->getText() ?? '';
+
+      if ($receiver is NameToken || $receiver is QualifiedName) {
+        if ($receiver is NameToken) {
+          $fn_name = $receiver->getText();
+        } else {
+          $receiver as QualifiedName;
+          $fn_name = '';
+          foreach ($receiver->getParts()->getChildren() as $part) {
+            invariant(
+              $part->getSeparator() is null ||
+                $part->getSeparator() is BackslashToken,
+              'Unexpected separator inside qualified function name: "%s"',
+              $part->getSeparatorx()->getText(),
+            );
+            $fn_name .= $part->getItem()?->getText() ?? '';
+            $fn_name .= $part->getSeparator()?->getText() ?? '';
+          }
         }
-      } else {  // probably a method call
-        continue;
-      }
 
-      $resolved_name = resolve_function($fn_name, $root, $node);
+        // Only replace calls to functions from the root namespace.
+        $resolved_name = resolve_function($fn_name, $root, $node);
+        $rule = self::getFunctionRules()[$resolved_name] ?? null;
 
-      $rule = self::getRules()[$resolved_name] ?? null;
+      } else if ($receiver is MemberSelectionExpression) {
+        if ($node->getArgumentList() is null) {
+          continue;
+        }
+
+        // Note that we don't care about the type of the object, only about the
+        // method name. This can result in some false positives, but that's OK
+        // because & and inout can be used interchangeably right now.
+        $rule = self::getMethodRules()[$receiver->getName()->getCode()] ?? null;
+
+      } else {
+        continue;
+      }
 
       if ($rule is nonnull) {
```

Comment:
with the `continue`, this should always be the case now?

Reply:
The `continue` is only for the "we don't know what this is" case (`$receiver` is neither of the 3 known types), this is still needed to handle the common case of a function/method for which there is no rule.